grpc/src/main/java/com/linecorp/armeria/server/grpc/GrpcServiceBuilder.java

```diff
@@ -491,9 +494,20 @@ public GrpcServiceBuilder exceptionMapping(GrpcStatusFunction statusFunction) {
      * Adds the specified exception mapping that maps a {@link Throwable} to a gRPC {@link Status}.
      * The mapping is used to handle a {@link Throwable} when it is raised.
      *
-     * <p>Note that this method and {@link #exceptionMapping(GrpcStatusFunction)} are mutually exclusive.
+     * <p>Note that this method and {@link #exceptionMapping(GrpcStatusFunction)} )} are mutually exclusive.
      */
     public GrpcServiceBuilder addExceptionMapping(Class<? extends Throwable> exceptionType, Status status) {
+        return addExceptionMapping(exceptionType, status, null);
+    }
+
+    /**
+     * Adds the specified exception mapping that maps a {@link Throwable} to a gRPC {@link Status}.
+     * The mapping is used to handle a {@link Throwable} when it is raised.
+     *
+     * <p>Note that this method and {@link #exceptionMapping(GrpcStatusFunction)} )} are mutually exclusive.
+     */
+    public <T extends Throwable> GrpcServiceBuilder addExceptionMapping(
+            Class<T> exceptionType, Status status, @Nullable Function<T, Metadata> metadataFunction) {
```

Comment:
Is it useful to take an exception for creating `Metadata`?
In this API, `exceptionType` is already specified. Is `Supplier<Metadata>` better for simplicity?

Reply:
Sometimes, we want to create a metadata from some values of a thrown exception.

e.g.

```kt
// thrown exception
BusinessLogicException extends RuntimeException {
  String message;
  String reason;
  String userId;
}
```

```proto
message RicherError {
  string message = 1;
  string reason = 2;
  string user_id = 3;
}
```